kit/Kit.hpp

```diff
@@ -186,6 +186,43 @@ class KitSocketPoll final : public SocketPoll
 #endif
 };
 
+class LogUiCmd
+{
+    struct lineOfUiLogFile
+    {
+        int iTime;
+        int iUser;
+        std::string aCmd;
+        std::string aSubCmd;
+        bool bMerged;
+        lineOfUiLogFile(int pTime, int pUser, std::string pCmd, std::string pSubCmd, bool pMerged)
+            : iTime(pTime)
+            , iUser(pUser)
+            , aCmd(std::move(pCmd))
+            , aSubCmd(std::move(pSubCmd))
+            , bMerged(pMerged)
+        {}
+    };
+public:
+    void createTmpFile();
+    void logLine(int userId, std::string line);
+    void processLogFile();
+    bool loadNextLine();
+    int findNextUserLine(int lineId);
+    bool mergeIfMouseClick(int lineId);
+    bool removeIfFalseUndoDecrease(int lineId);
+private:
+    // A temporary file for logging ui commands
+    std::fstream aFileStreamUICommands;
+    std::time_t aKitStartTimeSec;
+    std::string aKitStartTimeStr;
```

Comment:
Probably It could be optimized out by calculating it from aKitStartTimeSec somehow, but this way it was simple to do.